File: Algorithm/day11/P1.java
package Algorithm.day11;

import java.util.LinkedList;
import java.util.Queue;

public class P1 {
    public int solution(int[] queue1, int[] queue2) {
        int answer = 0;
        long sum = 0;
        long sum1 = 0;
        long sum2 = 0;
        Queue<Integer> q1 = new LinkedList<>();
        Queue<Integer> q2 = new LinkedList<>();
        for (int i : queue1) {
            q1.offer(i);
            sum += i;
            sum1 += i;
        }
        for (int i : queue2) {
            q2.offer(i);
            sum += i;
            sum2 += i;
        }
        if(sum % 2 == 1) return -1;
        while(answer<=600_000){
            if(sum1 == sum2) return answer;
            if(sum1 > sum2) {
                sum1 -= q1.peek();
                sum2 += q1.poll();
            }
            else{
                sum2 -= q2.peek();
                sum1 += q2.poll();
            }
            answer++;
        }
        return -1;
    }
}

File: Algorithm/programmers/p1.cpp
#include <bits/stdc++.h>
using namespace std;

int solution(vector<int> queue1, vector<int> queue2) {
    long long sum = 0;
    queue<int> q1, q2;
    long long s1 =0, s2 = 0;
    
    for(int i=0; i<queue1.size(); i++){
        sum += queue1[i] + queue2[i];
        
        q1.push(queue1[i]);
        s1 += queue1[i];
        
        q2.push(queue2[i]);
        s2 += queue2[i];
        
    }
    long long half = sum / 2;
    cout<<"s1:" <<s1;
    cout<<'\n';
    cout<<"s2:" <<s2;
    
    

    
    int mx = 900000;
    int cnt = 0;
    while(mx--){
        if(s1 == half) return cnt;
        if(q1.empty() || q2.empty()) break;
        if(s1 > half){
            cout<<q1.front()<<"가 q2으로"<<'\n';
            q2.push(q1.front());
            s2 += q1.front();
            s1 -= q1.front();
            q1.pop();
            
        }
        else{
            cout<<q2.front()<<"가 q1으로"<<'\n';
            q1.push(q2.front());
            s1 += q2.front();
            s2 -= q2.front();
            q2.pop();
        }
        cnt++;
    }
    
    return -1;
}